selftests/unit/test_archive.py

```diff
@@ -123,6 +124,17 @@ def get_path(*args):
         self.assertEqual(act, exp)
         self.assertEqual(os.path.realpath(get_path("link_to_dir")),
                          get_path("dir"))
+        # File permissions
+        self.assertEqual(os.stat(get_path("dir", "file2")).st_mode & 0o777,
```

Comment:
But, TBH, since this is pretty much about checking actual values on a file, and "0777" and the like are somewhat symbolic too, I'd be OK with non symbolic names.